cypress/integration/settings/username-change.js

```diff
@@ -0,0 +1,180 @@
+      cy.contains('Save').should('be.disabled');
+    });
+  });
+
+  it('Should not show anything if user types their current name', () => {
+    cy.get('@usernameInput')
+      .clear({ force: true })
+      .type('developmentuser', { force: true });
+  });
+
+  // eslint-disable-next-line
+  it('Should not be able to click the `Save` button if user types their current name', () => {
+    cy.get('@usernameInput')
+      .clear({ force: true })
+      .type('developmentuser', { force: true });
+
+    cy.get('@usernameForm').within(() => {
+      cy.contains('Save').should('be.disabled');
+    });
+  });
+
+  it('Should show warning if username includes invalid character', () => {
+    cy.get('@usernameInput')
+      .clear({ force: true })
+      .type('Quincy Larson', { force: true });
+
+    cy.contains('Username contains invalid characters.')
+      .should('be.visible')
+      .should('have.attr', 'role', 'alert')
+      .should('have.class', 'alert alert-danger');
+  });
+
+  // eslint-disable-next-line
+  it('Should not be able to click the `Save` button if username includes invalid character', () => {
+    cy.get('@usernameInput')
+      .clear({ force: true })
+      .type('Quincy Larson', { force: true });
+
+    cy.get('@usernameForm').within(() => {
+      cy.contains('Save').should('be.disabled');
+    });
+  });
+
+  it('Should change username if `Save` button is clicked', () => {
+    cy.get('@usernameInput')
+      .clear({ force: true })
+      .type('quincy', { force: true });
+
+    cy.get('@usernameForm').within(() => {
+      cy.wait(2000);
+      cy.contains('Save').click({ force: true });
+    });
+    cy.contains('Account Settings for quincy').should('be.visible');
+  });
+
+  it('Should show flash message showing username has been updated', () => {
+    cy.get('@usernameInput')
+      .clear({ force: true })
+      .type('nhcarrigan', { force: true });
+    cy.wait(2000);
+    cy.get('@usernameInput').type('{enter}', { force: true, release: false });
+
+    cy.contains('We have updated your username to nhcarrigan')
+      .should('be.visible')
+      .should(
+        'have.class',
+        'flash-message alert alert-success alert-dismissable'
+      );
+  });
+
+  it('Should be able to close the shown flash message', () => {
+    cy.get('@usernameInput')
+      .clear({ force: true })
+      .type('bjorno', { force: true });
+    cy.wait(2000);
+    cy.get('@usernameInput').type('{enter}', { force: true, release: false });
+
+    cy.contains('We have updated your username to bjorno').within(() => {
+      cy.get('button').click();
+    });
+
+    cy.contains('We have updated your username to bjorno').should(
+      'not.be.visible'
+    );
+  });
+
+  it('Should change username if enter is pressed', () => {
+    cy.get('@usernameInput')
+      .clear({ force: true })
+      .type('developmentuser', { force: true });
+    cy.wait(2000);
+    cy.get('@usernameInput').type('{enter}', { force: true, release: false });
```

Comment:
```suggestion
      .type('developmentuser', { force: true });
    cy.contains('Username is available.');
    cy.get('@usernameInput').type('{enter}', { force: true, release: false });
```
It should be possible to run each test in isolation and repeatedly, but this test assumes that the username is not `developmentuser`.  That's the case because the previous test set it to `bjorno`.  I think it would be more robust to normalise the settings in `beforeEach`.

Reply:
This seems to be tricky one... I'm trying a few things with this one. Sorry for the time it's taking. :sweat:

Reply:
There's absolutely nothing to apologise for.  You've already done a load of work on the Cypress testing and it's all incredibly useful!

So, for this particular point, I would try (if you haven't already) setting the username to be `developmentuser` `beforeEach` test and change this test to type in a different username.

Reply:
I achieved the same thing with a Cypress command.